src/libhoney.js

```diff
@@ -119,9 +119,24 @@ export default class Libhoney extends EventEmitter {
 
   _responseCallback(responses) {
     let queue = this._responseQueue;
-    if (queue.length < this._options.maxResponseQueueSize) {
+
+    //check that incoming responses can fit in under the max queue size
```

Comment:
wdyt about factoring this logic out to a separate function, something like `concatWithMaxLimit(arr1, arr2, limit)`.

logic/control flow is bound to be easier to follow when the `if (responsesFitUnderMax)` then clause can just return the result and therefore not need an `else `at all.

Reply:
I like it! I'll push up an update shortly